### PR TITLE
Hc merge frontend

### PR DESCRIPTION
**Pull request by SeePeaJay**

### CODE REVIEW

**Review by Huiwen-Zuo — DISMISSED** [review]

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Navbar/>
```
Comment (ddm1293) [inline frontend/src/App.js]: I feel like BrowserRouter should be placed here on top of the routes.

File: frontend/src/index.js [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
+      <Provider store={store}>
+        <Router>
```
Comment (ddm1293) [inline frontend/src/index.js]: Probably should be moved to app.js

File: frontend/src/components/Comments/CommentView.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    return (
+        <div className="comment-main">
+            <h3>Comments</h3>
+            <div className="comment-add">
+                <div><CommentInput/></div>
+            </div>
+            <div className="comment-list">
+                {comments.map((comment) => (
+                    <div className="comment-container" >
```
Comment (ddm1293) [inline frontend/src/components/Comments/CommentView.js]: how about extracting this to be a component?

File: frontend/src/components/Forum/Forum.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import { AppstoreOutlined, BarsOutlined, EnvironmentOutlined } from '@ant-design/icons'
+import '../../style/Forum.css'
+import { useSelector } from 'react-redux'
+import { useState } from 'react'
+import { threads } from '../../mocks/forumMock'
+
+const { Content } = Layout
+
+function Forum (props) {
+  //reference: ChatGPT (separate list of items into different pages)
```
Comment (ddm1293) [inline frontend/src/components/Forum/Forum.js]: please remove unnecessary comments, not limited to this one, but there are many unnecessary comments and commented out code.

File: frontend/src/App.js (hunk 1 above)
Comment (SeePeaJay) [reply frontend/src/App.js]: Ok, I can fix this

File: frontend/src/index.js (hunk 2 above)
Comment (SeePeaJay) [reply frontend/src/index.js]: Ok, I can fix this

File: frontend/src/components/Forum/Forum.js (hunk 4 above)
Comment (SeePeaJay) [reply frontend/src/components/Forum/Forum.js]: I have removed the ChatGPT reference part and what I considered to be unnecessary commented code.

File: frontend/src/components/Comments/CommentView.js (hunk 3 above)
Comment (SeePeaJay) [reply frontend/src/components/Comments/CommentView.js]: I'm sorry, but I'm not sure what exactly needs to be a component here, and why? Are you referring to everything inside and include the comment-list div?

Comment (ddm1293) [reply frontend/src/components/Comments/CommentView.js]: Because I feel like they are closedly knitted and are subject to future change?

**Review by ddm1293 — DISMISSED** [review]

File: frontend/src/components/Thread.js
Comment (Huiwen-Zuo) [inline frontend/src/components/Thread.js:1]: @SeePeaJay Hank could you please change the name tag of the username

>                 <h2 className="username">{user.username}</h2>

to something else like "thread-username" and the one in Thread.css as well so there won't be any display glitch? Thank you so much!

**Review by Hann61 — DISMISSED** [review]

**Review by Huiwen-Zuo — APPROVED** [review]
All good from my perspective, thank you!

**Review by ddm1293 — APPROVED** [review]

**Pull request merged**